lib/repo-url.js

```diff
@@ -0,0 +1,66 @@
+'use strict'
+
+const fs = require('fs')
+const path = require('path')
+const { URL } = require('url')
+const hostedGitInfo = require('hosted-git-info')
+
+/**
+ * @param packageName A package name as listed in package.json's dependencies list
+ * @param packageJson Optional param to specify a object representation of a package.json file instead of loading from node_modules
+ * @returns A valid url to the root of the package's source or null if a url could not be determined
+ */
+function getRepoUrl(packageName, packageJson = undefined) {
+  if (packageJson === undefined) {
+    try {
+      const nodeModulePaths = require.resolve.paths(packageName)
+      for (const basePath of nodeModulePaths) { // eslint-disable-line fp/no-loops
+        const packageJsonPath = path.join(basePath, packageName, 'package.json')
+        if (fs.existsSync(packageJsonPath)) {
+          packageJson = JSON.parse(fs.readFileSync(packageJsonPath, 'utf-8'))
+          break
+        }
+      }
+      // The package.json file cannot be found
+      if (packageJson === undefined) {
+        return null
+      }
+    }
+    catch (e) {
+      return null
+    }
```

Comment:
This logic should be factored out into a `getPackageRepo` function that returns `packageJson.repository` or `null`.